ccl/changefeedccl: TestChangefeedKafkaMessageTooLarge failed

ccl/changefeedccl.TestChangefeedKafkaMessageTooLarge [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8000745?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8000745?buildTab=artifacts#/) on master @ [7c8e7012b73a598f3aa1da9bbf422e833706f5d3](https://github.com/cockroachdb/cockroach/commits/7c8e7012b73a598f3aa1da9bbf422e833706f5d3):


```
=== RUN   TestChangefeedKafkaMessageTooLarge
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedKafkaMessageTooLarge2875385573
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedKafkaMessageTooLarge
    changefeed_test.go:7550: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedKafkaMessageTooLarge2875385573
--- FAIL: TestChangefeedKafkaMessageTooLarge (46.91s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka
    helpers_test.go:751: making server as secondary tenant
    helpers_test.go:837: making kafka feed factory
    --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka (46.90s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
    changefeed_test.go:7478: still waiting for job status; current reverting
        --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill (45.48s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92722 ccl/changefeedccl: TestChangefeedKafkaMessageTooLarge/kafka/eventually_surface_error_for_retry:_unrelated_error_mid-buffering failed [C-test-failure O-robot T-cdc branch-release-22.2]
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedKafkaMessageTooLarge.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-22552

Epic CRDB-11732

ccl/changefeedccl.TestChangefeedKafkaMessageTooLarge [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8101211?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8101211?buildTab=artifacts#/) on master @ [c4bde8b72cdd4016845ae70ef5162b3f11fab1fb](https://github.com/cockroachdb/cockroach/commits/c4bde8b72cdd4016845ae70ef5162b3f11fab1fb):


```
=== RUN   TestChangefeedKafkaMessageTooLarge
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedKafkaMessageTooLarge1541882091
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedKafkaMessageTooLarge
    changefeed_test.go:7551: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedKafkaMessageTooLarge1541882091
--- FAIL: TestChangefeedKafkaMessageTooLarge (46.53s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka
    helpers_test.go:754: making server as system tenant
    helpers_test.go:837: making kafka feed factory
    --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka (46.52s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
    changefeed_test.go:7479: still waiting for job status; current reverting
        --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill (45.39s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92722 ccl/changefeedccl: TestChangefeedKafkaMessageTooLarge/kafka/eventually_surface_error_for_retry:_unrelated_error_mid-buffering failed [C-test-failure O-robot T-cdc branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedKafkaMessageTooLarge.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedKafkaMessageTooLarge [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8601340?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8601340?buildTab=artifacts#/) on master @ [0d089058d1be30cc7fce61a898c63e2cf87795cd](https://github.com/cockroachdb/cockroach/commits/0d089058d1be30cc7fce61a898c63e2cf87795cd):


```
=== RUN   TestChangefeedKafkaMessageTooLarge
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/50a8284f7c7f32a812f6eaeb52caed36/logTestChangefeedKafkaMessageTooLarge1626955451
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedKafkaMessageTooLarge
    changefeed_test.go:7940: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/50a8284f7c7f32a812f6eaeb52caed36/logTestChangefeedKafkaMessageTooLarge1626955451
--- FAIL: TestChangefeedKafkaMessageTooLarge (47.73s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka
    helpers_test.go:810: making server as system tenant
    helpers_test.go:893: making kafka feed factory
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
    --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka (47.69s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
    changefeed_test.go:7868: still waiting for job status; current reverting
        --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill (45.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedKafkaMessageTooLarge.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedKafkaMessageTooLarge [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8696031?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8696031?buildTab=artifacts#/) on master @ [31365e21dc606cdc1e4302c86192ffc5a6cf1255](https://github.com/cockroachdb/cockroach/commits/31365e21dc606cdc1e4302c86192ffc5a6cf1255):


```
=== RUN   TestChangefeedKafkaMessageTooLarge
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedKafkaMessageTooLarge1934714961
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedKafkaMessageTooLarge
    changefeed_test.go:8036: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedKafkaMessageTooLarge1934714961
--- FAIL: TestChangefeedKafkaMessageTooLarge (46.62s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka
    helpers_test.go:810: making server as system tenant
    helpers_test.go:893: making kafka feed factory
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
    --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka (46.61s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
    changefeed_test.go:7964: still waiting for job status; current reverting
        --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill (45.36s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedKafkaMessageTooLarge.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedKafkaMessageTooLarge [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8750408?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8750408?buildTab=artifacts#/) on master @ [c95bef097bd4c213c6b5c0c125a9a846c4479d73](https://github.com/cockroachdb/cockroach/commits/c95bef097bd4c213c6b5c0c125a9a846c4479d73):


```
=== RUN   TestChangefeedKafkaMessageTooLarge
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedKafkaMessageTooLarge1213807325
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedKafkaMessageTooLarge
    changefeed_test.go:8036: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedKafkaMessageTooLarge1213807325
--- FAIL: TestChangefeedKafkaMessageTooLarge (49.50s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka
    helpers_test.go:807: making server as secondary tenant
    helpers_test.go:893: making kafka feed factory
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
    --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka (49.49s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
    changefeed_test.go:7964: still waiting for job status; current reverting
        --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill (45.89s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedKafkaMessageTooLarge.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedKafkaMessageTooLarge [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8771044?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8771044?buildTab=artifacts#/) on master @ [8124f66625a8451a5f45f1831246038fb0eeb2a1](https://github.com/cockroachdb/cockroach/commits/8124f66625a8451a5f45f1831246038fb0eeb2a1):


```
=== RUN   TestChangefeedKafkaMessageTooLarge
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b4b4ff567208b3b36d9c1ab53c36b00f/logTestChangefeedKafkaMessageTooLarge3619894655
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedKafkaMessageTooLarge
    changefeed_test.go:8037: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b4b4ff567208b3b36d9c1ab53c36b00f/logTestChangefeedKafkaMessageTooLarge3619894655
--- FAIL: TestChangefeedKafkaMessageTooLarge (47.97s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka
    helpers_test.go:810: making server as system tenant
    helpers_test.go:893: making kafka feed factory
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka
    testfeed_test.go:278: creating external connection
    testfeed_test.go:281: ran create external connection
    --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka (47.91s)
=== RUN   TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
=== CONT  TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill
    changefeed_test.go:7965: still waiting for job status; current reverting
        --- FAIL: TestChangefeedKafkaMessageTooLarge/kafka/succeed_against_a_large_backfill (45.63s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedKafkaMessageTooLarge.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/changefeedccl/testfeed_test.go
+++ b/pkg/ccl/changefeedccl/testfeed_test.go
@@ -1717,6 +1717,7 @@ func (s *fakeKafkaSink) Dial() error {
 				}
 				select {
 				case s.feedCh <- m:
+				case <-kafka.stopWorkerCh:
 				case <-s.tg.done:
 				}
 				return nil

